[CI] TransformSurvivesUpgradeIT testTransformRollingUpgrade failing

probably a regression of #80013

**Build scan:**
https://gradle-enterprise.elastic.co/s/toqidjvpswti4/tests/:x-pack:qa:rolling-upgrade:v7.11.2%23upgradedClusterTest/org.elasticsearch.upgrades.TransformSurvivesUpgradeIT/testTransformRollingUpgrade

**Reproduction line:**
`./gradlew ':x-pack:qa:rolling-upgrade:v7.11.2#upgradedClusterTest' -Dtests.class="org.elasticsearch.upgrades.TransformSurvivesUpgradeIT" -Dtests.method="testTransformRollingUpgrade" -Dtests.seed=11151805F4A613BA -Dtests.bwc=true -Dtests.locale=und -Dtests.timezone=America/Indiana/Knox -Druntime.java=8`

**Applicable branches:**
7.16

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.upgrades.TransformSurvivesUpgradeIT&tests.test=testTransformRollingUpgrade

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: a value greater than <7>
     but: <7> was equal to <7>

  at __randomizedtesting.SeedInfo.seed([11151805F4A613BA:BE682DBC5FE0C07E]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.upgrades.TransformSurvivesUpgradeIT.lambda$verifyContinuousTransformHandlesData$8(TransformSurvivesUpgradeIT.java:248)
  at org.elasticsearch.upgrades.TransformSurvivesUpgradeIT.lambda$awaitWrittenIndexerState$9(TransformSurvivesUpgradeIT.java:301)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1123)
  at org.elasticsearch.upgrades.TransformSurvivesUpgradeIT.awaitWrittenIndexerState(TransformSurvivesUpgradeIT.java:293)
  at org.elasticsearch.upgrades.TransformSurvivesUpgradeIT.verifyContinuousTransformHandlesData(TransformSurvivesUpgradeIT.java:243)
  at org.elasticsearch.upgrades.TransformSurvivesUpgradeIT.testTransformRollingUpgrade(TransformSurvivesUpgradeIT.java:153)
  at sun.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:498)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at java.lang.Thread.run(Thread.java:748)

```

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/transform/transforms/TransformCheckpointTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/transform/transforms/TransformCheckpointTests.java
@@ -30,8 +30,12 @@ import static org.hamcrest.Matchers.lessThan;
 public class TransformCheckpointTests extends AbstractSerializingTransformTestCase<TransformCheckpoint> {
 
     public static TransformCheckpoint randomTransformCheckpoint() {
+        return randomTransformCheckpoint(randomAlphaOfLengthBetween(1, 10));
+    }
+
+    public static TransformCheckpoint randomTransformCheckpoint(String transformId) {
         return new TransformCheckpoint(
-            randomAlphaOfLengthBetween(1, 10),
+            transformId,
             randomNonNegativeLong(),
             randomNonNegativeLong(),
             randomCheckpointsByIndex(),

--- a/x-pack/plugin/transform/src/internalClusterTest/java/org/elasticsearch/xpack/transform/persistence/TransformConfigManagerTests.java
+++ b/x-pack/plugin/transform/src/internalClusterTest/java/org/elasticsearch/xpack/transform/persistence/TransformConfigManagerTests.java
@@ -419,9 +419,9 @@ public class TransformConfigManagerTests extends TransformSingleNodeTestCase {
         );
 
         // add another old one, but not with an existing id
-        transformId = "transform_oldindex";
-        docId = TransformConfig.documentId(transformId);
-        transformConfig = TransformConfigTests.randomTransformConfig(transformId);
+        final String oldTransformId = "transform_oldindex";
+        docId = TransformConfig.documentId(oldTransformId);
+        transformConfig = TransformConfigTests.randomTransformConfig(oldTransformId);
 
         try (XContentBuilder builder = XContentFactory.jsonBuilder()) {
             XContentBuilder source = transformConfig.toXContent(builder, new ToXContent.MapParams(TO_XCONTENT_PARAMS));
@@ -431,18 +431,29 @@ public class TransformConfigManagerTests extends TransformSingleNodeTestCase {
             client().index(request).actionGet();
         }
 
-        transformIds.add(transformId);
+        // add a new checkpoint doc for the old transform to check id expansion ignores other documents, see gh#80073
+        assertAsync(
+            listener -> transformConfigManager.putTransformCheckpoint(
+                TransformCheckpointTests.randomTransformCheckpoint(oldTransformId),
+                listener
+            ),
+            true,
+            null,
+            null
+        );
+
+        transformIds.add(oldTransformId);
         assertAsync(listener -> transformConfigManager.getAllTransformIds(listener), transformIds, null, null);
         assertAsync(
             listener -> transformConfigManager.getAllOutdatedTransformIds(listener),
-            tuple(Long.valueOf(numberOfTransformsToGenerate + 1), Collections.singleton(transformId)),
+            tuple(Long.valueOf(numberOfTransformsToGenerate + 1), Collections.singleton(oldTransformId)),
             null,
             null
         );
 
         assertAsync(
             listener -> transformConfigManager.expandAllTransformIds(true, 10, listener),
-            tuple(Long.valueOf(numberOfTransformsToGenerate + 1), Collections.singleton(transformId)),
+            tuple(Long.valueOf(numberOfTransformsToGenerate + 1), Collections.singleton(oldTransformId)),
             null,
             null
         );

--- a/x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/IndexBasedTransformConfigManager.java
+++ b/x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/IndexBasedTransformConfigManager.java
@@ -882,6 +882,10 @@ public class IndexBasedTransformConfigManager implements TransformConfigManager 
             .setSize(page.getSize())
             .setFetchSource(false)
             .addDocValueField(TransformField.ID.getPreferredName())
+            .setQuery(
+                QueryBuilders.boolQuery()
+                    .filter(QueryBuilders.termQuery(TransformField.INDEX_DOC_TYPE.getPreferredName(), TransformConfig.NAME))
+            )
             .request();
 
         executeAsyncWithOrigin(

